server/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -188,18 +143,10 @@ public static Checkpoint read(Path path) throws IOException {
             try (IndexInput indexInput = dir.openInput(path.getFileName().toString(), IOContext.DEFAULT)) {
                 // We checksum the entire file before we even go and parse it. If it's corrupted we barf right here.
                 CodecUtil.checksumEntireFile(indexInput);
-                final int fileVersion = CodecUtil.checkHeader(indexInput, CHECKPOINT_CODEC, INITIAL_VERSION, CURRENT_VERSION);
-                if (fileVersion == INITIAL_VERSION) {
-                    assert indexInput.length() == V1_FILE_SIZE : indexInput.length();
-                    return Checkpoint.readCheckpointV5_0_0(indexInput);
-                } else if (fileVersion == VERSION_6_0_0) {
-                    assert indexInput.length() == V2_FILE_SIZE : indexInput.length();
-                    return Checkpoint.readCheckpointV6_0_0(indexInput);
-                } else {
-                    assert fileVersion == CURRENT_VERSION : fileVersion;
-                    assert indexInput.length() == V3_FILE_SIZE : indexInput.length();
-                    return Checkpoint.readCheckpointV6_4_0(indexInput);
-                }
+                final int fileVersion = CodecUtil.checkHeader(indexInput, CHECKPOINT_CODEC, CURRENT_VERSION, CURRENT_VERSION);
+                assert fileVersion == CURRENT_VERSION : fileVersion;
+                assert indexInput.length() == V3_FILE_SIZE : indexInput.length();
+                return Checkpoint.readCheckpointV6_4_0(indexInput);
```

Comment:
Suggest removing the mention of a specific (ancient) version here too:

```suggestion
                return Checkpoint.readCheckpointV3(indexInput);
```